tests/main/snap-quota-cpu/task.yaml

```diff
@@ -0,0 +1,129 @@
+summary: Functional test for cpu quota-related snap commands.
+
+details: |
+  Functional test for snap cpu quota group commands ensuring that they are 
+  effective in practice.
+
+# these systems do not support quota groups due to their old systemd versions,
+# we do have a check to do for these systems, but that is just to ensure that
+# we don't allow using quota groups, and that check is done in the snap-quota
+# spread instead
+systems:
+  - -ubuntu-14.04-*
+  - -centos-7-*
+  - -amazon-linux-2-*
+  - -ubuntu-16.04-*
+  - -ubuntu-core-16-*
+
+prepare: |
+  snap install test-snapd-stressd --edge --devmode
+  snap set system experimental.quota-groups=true
+  tests.cleanup defer snap unset system experimental.quota-groups
+
+execute: |
+  echo "Starting service and verifying cpu usage"
+  snap start test-snapd-stressd.stress-sc
+
+  # helper function that has retry logic to allow for cpu usage to stabilize
+  function ensure_cpu_usage_higher_than {
+    n=0
+    until [ "$n" -ge 5 ]
+    do
+      usage=$(top -b -n 2 -d 0.2 | head | grep "$1" | awk '{print $9*100}')
+      if [ "$usage" -gt "$2" ]; then
+        return 0
+      fi
+      n=$((n+1))
+      sleep 1
+    done
+
+    echo "Expected cpu usage to be minimum $2%, got $usage"
+    exit 1
+  }
+  
+  echo "Verifying cpu usage, expecting atleast above 60% as we test against 50%"
+  ensure_cpu_usage_higher_than "stress" 6000
```

Comment:
(nitpick) I think it would feel slightly more natural here if you pass in real percentage (`60`) here and just expand inside the ensure_cpu_usage_higher_than() as it's a bit of an implementation detail that bash is silly and needs integers.

Reply:
Yea that makes sense, i'll get that changed, good point